extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/extractors/SoundcloudStreamExtractor.java

```diff
@@ -185,43 +187,102 @@ public String getHlsUrl() {
         try {
             final JsonArray transcodings = track.getObject("media").getArray("transcodings");
 
-            // get information about what stream formats are available
-            for (Object transcoding : transcodings) {
+            // Iterate a first time to see if there is a progressive MP3 stream available.
+            // If yes, the MP3 HLS stream will be not added to audioStreams.
 
+            boolean mp3ProgressiveStreamInTranscodings = false;
+
+            for (final Object transcoding : transcodings) {
+                final JsonObject t = (JsonObject) transcoding;
+                if (t.getString("preset").contains("mp3") &&
+                    t.getObject("format").getString("protocol").equals("progressive")) {
```

Comment:
Another minor that i personally think helps readability is seperating the parts where json is parsed and the part where the comparison is made. Inside the if I don't need to also know how the data was extracted in order to understand if it should go into the if statement

```java
String preset = t.getString("preset");
String protocol = t.getObject("format").getString("protocol");
if (preset.contains("mp3") && protocol.equals("progressive")) {
```

Some performance is lost because `protocol` is always evaluated unlike before and more objects are created. but it's easier to read. And maybe there is some way to remove that downside, that i can't think of right now.
Whether that tradeoff is worth it I can't say, especially since we are on Android where resources are a lot more previous compared to other plattforms

Reply:
A lot more previous?

Reply:
precious!!!

stupid github. Whenever i start reviewing and edit a review comment before submitting it, it does not save the edit